index.js

```diff
@@ -1,117 +1,142 @@
- author:'J.R.R. Tolkien',
- director: 'this person',
- genre: 'this genre'
-},
-{
- id: 3,
- title: 'Twilight',
- author: 'stephanie Meyer',
- director: 'this person',
- genre: 'this genre'
- }
-];
+app.use(morgan("common"));
+app.use(bodyParser.json());
 
-app.use(morgan("common")); 
 app.use(express.static("public"));
 
-// gets the data about a single director, by name \\
+let topMovies = [
+  {
+    id: 1,
+    title: "harry potter and the sorcerer's stone",
+    author: "J.K. Rowling",
+    director: "this person",
+    genre: "this genre",
+  },
+  {
+    id: 2,
+    title: "Lord of the rings",
+    author: "J.R.R. Tolkien",
+    director: "this person",
+    genre: "this genre",
+  },
+  {
+    id: 3,
+    title: "Twilight",
+    author: "stephanie Meyer",
+    director: "this person",
+    genre: "this genre",
+  },
+];
 
-app.get('/movies/:director', (req, res) => {
-  res.json(topMovies.find((movies) => {
-  return movies.director ===req.params.director
-  }));
-  });
-  
-  app.get('/movies/:director/:genre', (req, res) => {
-    res.json(topMovies.find((movies) => {
-    return movies.genre ===req.params.genre
-    }));
-    });
-  
-  //adds data for a new movie to our list of movies. \\
-  
-  app.post('/movies', (req, res) => {
+app.get("/movies", (req, res) => {
+  res.json(topMovies);
+});
+
+//Get a single movie by title
+app.get("/movies/:title", (req, res) => {
+  res.json(
+    topMovies.find((movies) => {
+      return movies.title === req.params.title;
+    })
+  );
+});
+
+//Get list of movies by director
+app.get("/movies/director/:director", (req, res) => {
+  res.json(
+    topMovies.filter((movies) => {
+      return movies.director === req.params.director;
+    })
+  );
+});
+
+//Get list of movies by genre
+app.get("/movies/genre/:genre", (req, res) => {
+  res.json(
+    topMovies.filter((movies) => {
+      return movies.genre === req.params.genre;
+    })
+  );
+});
+
+//adds data for a new movie to our list of movies. \\
+app.post("/movies", (req, res) => {
   let newMovie = req.body;
-  
+
   if (!newMovie.name) {
-  const message = 'missing name in request body';
-  res.status(400).send(message);
+    const message = "missing name in request body";
+    res.status(400).send(message);
   } else {
-  newMovie.id = uuid.v4();
-  movies.push(newMovie);
-  res.status(201).send(newMovie);
-   }
-  });
-  
-  // deletes a movie from our list by ID \\
-  app.delete('/movies/:id', (req, res) => {
+    newMovie.id = uuid.v4();
+    movies.push(newMovie);
+    res.status(201).send(newMovie);
+  }
+});
+
+// deletes a movie from our list by ID \\
+app.delete("/movies/:id", (req, res) => {
   let movie = topMovies.find((movie) => {
-  return movie.id === req.params.id
+    return movie.id === req.params.id;
   });
-  
+
   if (movie) {
-  movie = movie.filter((obj) => {
-  return obj.id !== req.params.id 
-  });
-  res.status(201).send('movie ' + req.params.id + ' was deleted');
-  
-  }
-  });
```

Comment:
Not a lot of changes here, just updated `movie = movie.filter` to `topMovies = topMovies.filter` since we want to remove the movie from the array.